chips/arty_e21_chip/src/chip.rs

```diff
@@ -134,11 +121,8 @@ impl<'a, I: InterruptService<()> + 'a> ArtyExx<'a, I> {
             addi t0, t0, %lo(_start_trap)
             ori  t0, t0, 0x02 // Set CLIC direct mode
             csrw 0x305, t0    // Write the mtvec CSR.
-        "
-        :
-        :
-        :
-        : "volatile");
+            "
```

Comment:
This is more a general question: shouldn't we mark the register `t0` as clobbered here (as described [here](https://doc.rust-lang.org/beta/unstable-book/library-features/asm.html#clobbered-registers))?

Given that this entire function is a large assembly block, it's probably fine, but I imagine the compiler could potentially do some optimizations or perform inlining which could break the assumptions made about registers not being clobbered in the future?

Reply:
I agree, t0 should be marked as clobbered.